src/main/scala/mesosphere/marathon/MarathonConf.scala

```diff
@@ -307,5 +307,12 @@ trait MarathonConf
     noshort = true,
     hidden = true,
     default = Some(MesosHeartbeatMonitor.DEFAULT_HEARTBEAT_FAILURE_THRESHOLD))
+
+  lazy val drainingTime = opt[Long](
+    "draining_time",
```

Comment:
Firstly, thanks for the work on this @alanbover!

I'm following this PR closely because we make use of Mesos maintenance primitives with https://github.com/Yelp/paasta

Currently when we schedule a host for maintenance and it enters a draining state we proactively try and scale up our Marathon apps to ensure that we have a new task somewhere else before we kill the draining host. However, as you know this leaves a possibility that Marathon will schedule the host back on the same box. Currently, we use http://mesos.readthedocs.io/en/0.23.1/reservation/#dynamic-reservation-since-0230 to reserve the resources to minimise the likelihood of this happening.

However, this PR would be a step in the right direction because we would eliminate that possibility if Marathon just dropped offers from maint'd hosts. I think from our point of view I'd expect Marathon to drop all offers from hosts scheduled for maintenance (i.e. draining) as well as any hosts currently in maintenance.

I hope that is useful context? And thanks again for looking at this!